File: 0111_Minimum Depth of Binary Tree/MinimumDepthofBinaryTree.cs
public class Solution {
    public int MinDepth(TreeNode root) {
        if(root == null) return 0;
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        
        var level = 0;
        while(queue.Count > 0)
        {
            level++;
            var c = queue.Count;
            while(c-- > 0)
            {
                var n = queue.Dequeue();
                if(n.left == null && n.right == null) return level;
                if(n.left!=null) queue.Enqueue(n.left);
                if(n.right!=null) queue.Enqueue(n.right);
            }
        }
        
        return -1;
    }
}

File: 0111_Minimum Depth of Binary Tree/MinimumDepthofBinaryTree.java
class Solution {
    public int minDepth(TreeNode root) {
    	if(root == null) return 0;
    	int left = minDepth(root.left);
    	int right = minDepth(root.right);
    	return (left == 0 || right == 0) ? left + right + 1 : Math.min(left, right) + 1;
    }
   
    /**
     * Another solution.
     * 
    public int minDepth(TreeNode root) {
    	if (root == null) return 0;
    	int left = minDepth(root.left)
    	int right = minDepth(root.right);
    	return 1 + (Math.min(left, right) > 0 ? Math.min(left, right) : Math.max(left, right));
	}
     */
}
